### ISSUE TITLE
[Accessibility] [Search] Search component results listbox missing label and it's children missing required role.

### ISSUE THREAD

**Issue by CameronSharp1980** [adobe/aem-core-wcm-components#1500]
## Bug Report

**Current Behavior**
The Search component results listbox has the role="listbox" attribute but it's children do not have the role="option" attribute or aria-selected attributes as required. The listbox is also missing a label, labelledby or title attribute.

**Expected behavior/code**
The element with the role="listbox" attribute should be populated with child elements containing the role="option" attribute and these children should have the aria-selected attribute set to true and false when they are selected or not. If none are selected the aria-selected attribute can be omitted. The listbox element should also have an aria-label, aria-labelledby or title attribute for screen readers.

**Environment**
- AEM version and patch level: 6.5.8.0 & AEM as a cloud service
- Core Components version: 2.15.0
- JRE version: Java(TM) SE Runtime Environment (build 1.8.0_241-b07)

**Possible Solution**
Add the role="option" role to the child elements of the listbox and give the listbox an aria-labelledby attribute pointing to it's label. Add toggling behavior to these role="option" child elements toggling aria-selected to true/false when they are selected or not and removing the attribute entirely when none are selected.

**Additional context / Screenshots**
More details about the aria listbox role and it's requirements can be found here:
 - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/listbox_role

Screenshots of issue from Axe tool:
![Screen Shot 2021-04-06 at 1 38 16 PM](https://user-images.githubusercontent.com/32853688/113768948-abd90f00-96dd-11eb-90ff-427239711ade.png)

![Screen Shot 2021-04-06 at 1 39 48 PM](https://user-images.githubusercontent.com/32853688/113769028-be534880-96dd-11eb-9462-6624a9bd82b0.png)
